internal/network/command_private_link_attachment_connection.go

```diff
@@ -51,9 +54,25 @@ func printPrivateLinkAttachmentConnectionTable(cmd *cobra.Command, connection ne
 		Phase:                   connection.Status.GetPhase(),
 	}
 
-	if connection.Spec.Cloud != nil && connection.Spec.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnection != nil {
-		out.Cloud = CloudAws
-		out.AwsVpcEndpointId = connection.Spec.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnection.GetVpcEndpointId()
+	if connection.Spec.HasCloud() {
+		switch {
+		case connection.Spec.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnection != nil:
+			out.Cloud = CloudAws
+			out.AwsVpcEndpointId = connection.Spec.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnection.GetVpcEndpointId()
+		case connection.Spec.Cloud.NetworkingV1AzurePrivateLinkAttachmentConnection != nil:
+			out.Cloud = CloudAzure
+			out.AzurePrivateEndpointResourceId = connection.Spec.Cloud.NetworkingV1AzurePrivateLinkAttachmentConnection.GetPrivateEndpointResourceId()
+		}
+	}
+
+	if connection.Status.HasCloud() {
+		switch {
+		case connection.Status.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus != nil:
+			out.AwsVpcEndpointServiceName = connection.Status.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus.GetVpcEndpointServiceName()
+		case connection.Status.Cloud.NetworkingV1AzurePrivateLinkAttachmentConnectionStatus != nil:
+			out.AzurePrivateLinkServiceAlias = connection.Status.Cloud.NetworkingV1AzurePrivateLinkAttachmentConnectionStatus.GetPrivateLinkServiceAlias()
+			out.AzurePrivateLinkServiceId = connection.Status.Cloud.NetworkingV1AzurePrivateLinkAttachmentConnectionStatus.GetPrivateLinkServiceResourceId()
+		}
 	}
 
 	if connection.Status.Cloud != nil && connection.Status.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus != nil {
```

Comment:
do we still need this clause?
`if connection.Status.Cloud != nil && connection.Status.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus != nil`

Reply:
will remove!